src/middlewares/loggers/api.logger.ts

```diff
@@ -0,0 +1,22 @@
+import { Request, Response, NextFunction } from 'express';
+
+const apiLogger = (req: Request, res: Response, next: NextFunction) => {
+    const { body, originalUrl, method } = req;
+
+    const oldSend = res.send;
```

Comment:
Could you give me a source of this approach?
At the moment, it looks to me as if the "res.send" is not deeply copied (hence modified down the line cause it's 
an object and passed by reference) and also response is terminated with "res.send" at some point.

Reply:
Sure, I adapted it from the answer here: https://stackoverflow.com/a/69549821

This was largely done out of frustration not being able to receive the response body message with either the `json()` or body-parser middlewares (I'll confess this might have been a lack of understanding on my part though - everywhere I looked it seemed to tell me to avoid the latter as it's depreciated). Given your explanation I can see this might not be the best approach.

Other answers in the same SO post and elsewhere either didn't work for me or play nicely with TypeScript. Might it be easier to use a library like [morgan-body](https://github.com/sirrodgepodge/morgan-body)?

Reply:
Oh, I can see now.
It's not typically done in production since the response body is already available before we send it compared to the incoming requests ;)
My suggestion is to try "morgan-body" or drop the functionality if the package doesn't help.

Reply:
So, morgan-body worked fine, but I felt the customisation options were severely lacking for logging (And in the end I started to wonder if adding another dep might render the logger service somewhat redundant as it would only be logging time at that point). Plus it interfered with the way I had my formatting options organised. 

If the response body isn't really necessary I'm happy to skip this - otherwise if this might affect the completion success criteria I'll take another crack at this. Just let me know :).

Reply:
Happy you've played with the logging stuff a bit more anyway :)
It's often overlooked by devs because of the amount of features offered out-of-the-box by logging libraries. And a solid understanding why we log the way we do and what are the options help A LOT when there's more than 1 backend service.